buffer/src/main/java/io/netty/buffer/PooledSlicedByteBuf.java

```diff
@@ -114,17 +114,14 @@ public ByteBuf retainedSlice(int index, int length) {
     @Override
     public ByteBuf duplicate() {
         // Capacity is not allowed to change for a sliced ByteBuf, so length == capacity()
-        final ByteBuf duplicate = unwrap().slice(adjustment, capacity());
+        final ByteBuf duplicate = super.slice(adjustment, capacity());
```

Comment:
Using a slice was here before this PR ... and I'm guessing was done to be consistent with the unpooled derived buffers. However you triggered me to look deeper into this topic and I don't think it is correct to use a slice in this situation. One reason is duplicate/slice buffers have different behavior around capacity and max capacity. Duplicate allows setting the capacity while sliced buffers do not. There may also be other differences and it seems more straightforward to use a duplicate object when the user requests it. I added some unit tests for the pooled derived buffers and I will submit a followup PR for the unpooled derived buffers.